client/cherami/client.go

```diff
@@ -85,10 +85,29 @@ func NewHyperbahnClient(serviceName string, bootstrapFile string, options *Clien
 	return newClientWithTChannel(ch, options)
 }
 
-// NewClientWithFE is used by Frontend to create a Cherami client for itself.
-// It is used by non-streaming publish/consume APIs.
+// NewClientWithFEClient is used by Cherami Frontend to create a Cherami client for itself.
+// Cherami customers shouldn't need to use this function.
+func NewClientWithFEClient(feClient cherami.TChanBFrontend, options *ClientOptions) (Client, error) {
```

Comment:
Can we add this new API when we need it  ? my concern is, we add this new API and never update the callers to migrate from old to new. Since this API is only for internal, its reasonable to even panic if the options is invalid. 

Alternatively, we can just delete the old API, that will force the caller to change the code when the dependencies are upgraded.